### PR TITLE
DSND-3006: Add mongo versioning to upsert flow

### PR DESCRIPTION
**Pull request by crowlandsCH**
* adds versioning to documents persisted within the company-profile collection to prevent concurrency issues. 
* All pom dependencies updated and unnecessary ones cleaned. 
* Code needs a more major refactor at a later date and persistence models need changing. 

[DSND-3006](https://companieshouse.atlassian.net/browse/DSND-3006)

[DSND-3006]: https://companieshouse.atlassian.net/browse/DSND-3006?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by ch-code-analysis** [comment]
CI: Security warnings found!

**Comment by ch-code-analysis** [comment]
CI: Security warnings found!

File: src/itest/java/uk/gov/companieshouse/company/profile/controller/CompanyProfileControllerITest.java [hunk 1]
```diff
@@ -51,7 +51,7 @@ void getCompanyProfileWithMatchingCompanyNumber() throws Exception {
         //Updated updated = mock(Updated.class);
```
Comment (cgregoryCH) [inline src/itest/java/uk/gov/companieshouse/company/profile/controller/CompanyProfileControllerITest.java:51]: Can this comment and unused variable above be removed?

File: src/main/java/uk/gov/companieshouse/company/profile/model/UnversionedCompanyProfileDocument.java [hunk 2]
```diff
@@ -0,0 +1,52 @@
+package uk.gov.companieshouse.company.profile.model;
+
+import uk.gov.companieshouse.api.model.CompanyProfileDocument;
+
+public class UnversionedCompanyProfileDocument  extends CompanyProfileDocument {
+
+    private Long version;
+
+    UnversionedCompanyProfileDocument() {
```
Comment (cgregoryCH) [inline src/main/java/uk/gov/companieshouse/company/profile/model/UnversionedCompanyProfileDocument.java:9]: Can be removed

File: src/main/java/uk/gov/companieshouse/company/profile/model/UnversionedCompanyProfileDocument.java [hunk 3]
```diff
@@ -0,0 +1,52 @@
+package uk.gov.companieshouse.company.profile.model;
+
+import uk.gov.companieshouse.api.model.CompanyProfileDocument;
+
+public class UnversionedCompanyProfileDocument  extends CompanyProfileDocument {
```
Comment (cgregoryCH) [inline src/main/java/uk/gov/companieshouse/company/profile/model/UnversionedCompanyProfileDocument.java:5]: extra space before extends

File: src/test/java/uk/gov/companieshouse/company/profile/transform/CompanyProfileTransformerTest.java [hunk 4]
```diff
@@ -1,25 +1,28 @@
...
 import java.time.LocalDateTime;
 import java.time.ZoneOffset;
 import java.time.format.DateTimeFormatter;
 
-public class CompanyProfileTransformerTest {
+class CompanyProfileTransformerTest {
 
     Logger logger = new StructuredLogger("CompanyProfileTransformerTest");
 
+    private TestHelper testHelper;
```
Comment (cgregoryCH) [inline src/test/java/uk/gov/companieshouse/company/profile/transform/CompanyProfileTransformerTest.java:25]: Can be removed

**Review by cgregoryCH — COMMENTED** [review]
Looks good, just a few nitpicks

**Comment by ch-code-analysis** [comment]
CI: Security warnings found!

File: src/test/java/uk/gov/companieshouse/company/profile/transform/CompanyProfileTransformerTest.java (hunk 4 above)
Comment (crowlandsCH) [reply src/test/java/uk/gov/companieshouse/company/profile/transform/CompanyProfileTransformerTest.java]: resolved in most recent commit

File: src/main/java/uk/gov/companieshouse/company/profile/model/UnversionedCompanyProfileDocument.java (hunk 3 above)
Comment (crowlandsCH) [reply src/main/java/uk/gov/companieshouse/company/profile/model/UnversionedCompanyProfileDocument.java]: resolved in most recent commit

File: src/main/java/uk/gov/companieshouse/company/profile/model/UnversionedCompanyProfileDocument.java (hunk 2 above)
Comment (crowlandsCH) [reply src/main/java/uk/gov/companieshouse/company/profile/model/UnversionedCompanyProfileDocument.java]: resolved in most recent commit

File: src/itest/java/uk/gov/companieshouse/company/profile/controller/CompanyProfileControllerITest.java (hunk 1 above)
Comment (crowlandsCH) [reply src/itest/java/uk/gov/companieshouse/company/profile/controller/CompanyProfileControllerITest.java]: resolved in most recent commit

**Comment by ch-code-analysis** [comment]
CI: Security warnings found!

**Review by sthompsonCH — APPROVED** [review]

**Pull request merged**